common/config/src/lib.rs

```diff
@@ -1,6 +1,8 @@
+#![no_std]
+
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Default)]
```

Comment:
Deriving `Default` is not what we want here because it will initialize the logging fields with `false` (it doesn't consider the serde-specific attributes). What I meant in https://github.com/rust-osdev/bootloader/pull/326#discussion_r1087736663 was the following:

- keep the manual `Default` implementation
- set the [`serde(default)` attribute](https://serde.rs/container-attrs.html#default) for the `BootConfig` struct
  - This attribute fills any missing fields from the `Default` implementation of the struct.
- remove the `serde(default)` attributes from the struct fields, as they're no longer needed
  - This also removes the need for the `default_logger_status` function.

Reply:
Thanks! I didn't understand it. Now, the code is cleaner.